tools/bcdb_mock/nodes_store.go

```diff
@@ -134,14 +134,15 @@ func (s *nodeStore) updateUptime(nodeID string, uptime int64) error {
 	return nil
 }
 
-func (s *nodeStore) StoreProof(nodeID string, dmi dmi.DMI, disks capacity.Disks) error {
+func (s *nodeStore) StoreProof(nodeID string, dmi dmi.DMI, disks capacity.Disks, hypervisor []string) error {
 	node, err := s.Get(nodeID)
 	if err != nil {
 		return err
 	}
 
 	proof := directory.TfgridNodeProof1{
-		Created: schema.Date{Time: time.Now()},
+		Created:    schema.Date{Time: time.Now()},
+		Hypervisor: hypervisor,
```

Comment:
Actually this is not enough. You will see in the code down, that it is possible that we do not store the `proof` variable:
https://github.com/threefoldtech/zos/blob/a7a8e9b6ea2a507f4e648c2d75cbb576bc0d6ae6/tools/bcdb_mock/nodes_store.go#L170-L174
I would replace this block with
```go
// don't save the proof if we already have one with the same
	// hash/content
	for i := range node.Proofs {
		if proof.Equal(node.Proofs[i]) {
			// always update hypervisor content
			node.Proofs[i].Hypervisor = hypervisor
			return nil
		}
	}
```
So we ensure that we always update the content of hypervisor even if the proof is already known and we do not store a new one.